### PR TITLE
[BUG] Modified VAR code to allow predict_quantiles of 0.5 (fixes #4742)

### LINKED ISSUE #4742: [BUG] `VAR` cannot predict quantiles when `alpha` is `0.5`
**Describe the bug**

`VAR` is unable to predict `0.5` quantile using `predict_quantiles`, i.e. median.

**To Reproduce**

```python
from sktime.datasets import load_longley
_, y = load_longley()

from sktime.forecasting.var import VAR
forecaster = VAR().fit(y)

forecaster.predict_quantiles(fh=[1, 2, 3], alpha=[0.025, 0.5, 0.975])
```

**Expected behavior**

No error. Currently it fails with the following:

> ValueError: alpha must be between 0 and 1

**Additional context**

I understand this should be same as `predict`, but I think it should still be able to predict as quantile nevertheless. `0.5` seems to work for `AutoETS`, `AutoARIMA`, `StatsForecastAutoARIMA`.

**Versions**
<details>

```pycon
>>> from sktime import show_versions; show_versions()

System:
    python: 3.9.16 (main, Mar  8 2023, 14:00:05)  [GCC 11.2.0]
executable: /path/to/venv/bin/python
   machine: Linux-5.15.90.1-microsoft-standard-WSL2-x86_64-with-glibc2.35

Python dependencies:
          pip: 23.1.2
       sktime: 0.19.2
      sklearn: 1.2.2
       skbase: 0.4.0
        numpy: 1.24.3
        scipy: 1.10.1
       pandas: 1.5.3
   matplotlib: 3.7.1
       joblib: 1.2.0
  statsmodels: 0.13.5
        numba: 0.57.1
     pmdarima: 2.0.3
      tsfresh: 0.20.0
   tensorflow: 2.12.0
tensorflow_probability: 0.19.0
>>> 
```

</details>

### PR DESCRIPTION
**Pull request by meraldoantonio**
#### Reference Issues/PRs
Fixes #4742

#### What does this implement/fix? Explain your changes.
Previously, VAR produced an error when predicting quantiles with an alpha of 0.5. This issue arose because alpha was converted to a coverage of 1.0 in `_base`, which was then reconverted to an alpha ("internal alpha") of 1.0 inside VAR's `_predict_interval`. An internal alpha of 1.0 is outside the acceptable range (0.0 to 1.0 exclusive) for statsmodels VAR.

This change hackily coerces an internal alpha of 1.0 to 0.99999 to avoid this error.

#### Does your contribution introduce a new dependency? If yes, which one?
No

#### What should a reviewer concentrate their feedback on?
The hacky nature of this change

#### Did you add any tests for the change?
No

##### For all contributions
- [X ] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/CONTRIBUTORS.md) with any new badges I've earned :-)
  How to: add yourself to the [all-contributors file](https://github.com/sktime/sktime/blob/main/.all-contributorsrc) in the `sktime` root directory (not the `CONTRIBUTORS.md`). Common badges: `code` - fixing a bug, or adding code logic. `doc` - writing or improving documentation or docstrings. `bug` - reporting or diagnosing a bug (get this plus `code` if you also fixed the bug in the PR).`maintenance` - CI, test framework, release.
  See here for [full badge reference](https://allcontributors.org/docs/en/emoji-key)
- [ ] Optionally, for added estimators: I've added myself and possibly to the `maintainers` tag - do this if you want to become the owner or maintainer of an estimator you added.
  See here for further details on the [algorithm maintainer role](https://www.sktime.net/en/latest/get_involved/governance.html#algorithm-maintainers).
- [ X] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

### CODE REVIEW

File: sktime/forecasting/var.py [hunk 1]
```diff
@@ -245,7 +245,9 @@ def _predict_interval(self, fh, X, coverage):
 
         for cov in coverage:
             alpha = 1 - cov
-
+            # A hacky way to coerce error-inducing alpha==1 into its approximant
+            if alpha == 1:
```
Comment (fkiraly) [inline sktime/forecasting/var.py:249]: I would do `if alpha >= 0.9999` or similar

Comment (meraldoantonio) [reply sktime/forecasting/var.py]: Sure, I've modified it to `if alpha >= 0.99999`

File: .all-contributorsrc
Comment (yarnabrina) [inline .all-contributorsrc:1]: You do not need to add this in each PR, only for the first time. You can of course add a new badge in the existing entry.

File: sktime/regression/deep_learning/cnn.py
Comment (yarnabrina) [inline sktime/regression/deep_learning/cnn.py:1]: I think this is coming from your old PR, and based on your branch it's very outdated from our `main` branch. Can you please update your fork with `sktime:main`, and then make the bug fix?

File: sktime/forecasting/var.py [hunk 2]
```diff
@@ -245,7 +245,9 @@ def _predict_interval(self, fh, X, coverage):
 
         for cov in coverage:
             alpha = 1 - cov
-
+            # A hacky way to coerce error-inducing alpha==1 into its approximant
+            if alpha >= 0.99999:
+                alpha = 0.99999
```
Comment (yarnabrina) [inline sktime/forecasting/var.py:250]: Thoughts: can/should we document this hack for users, and if so, where?

**Review by yarnabrina — CHANGES_REQUESTED** [review]

File: sktime/forecasting/var.py (hunk 2 above)
Comment (fkiraly) [reply sktime/forecasting/var.py:250]: docstring?

Comment (yarnabrina) [reply sktime/forecasting/var.py:250]: I think docstring modifications in these file do not get shown in read the docs. These are private methods (starting with _), and what all gets shown is only those of the base class.

Please let me know if this a misunderstanding.

Comment (fkiraly) [reply sktime/forecasting/var.py:250]: I mean, the class docstring, that gets overridden.

File: sktime/regression/deep_learning/cnn.py
Comment (meraldoantonio) [reply sktime/regression/deep_learning/cnn.py:1]: Yes indeed it's from my old PR, sorry! 
Sure, I'll update my fork and then make the bug fix

File: .all-contributorsrc [hunk 3]
```diff
@@ -2649,7 +2649,8 @@
       "avatar_url": "https://avatars.githubusercontent.com/u/37468543?v=4",
       "profile": "https://github.com/meraldoantonio",
       "contributions": [
-        "doc"
+        "doc",
+        "bug"
```
Comment (fkiraly) [inline .all-contributorsrc:2653]: `"bug"` is for diagnosing or reporting bugs, but not bugfixes. Bugfixes fall under the `"code"` badge, so you can add both.

**Review by yarnabrina — APPROVED** [review]
Approved in principle, but following two changes are strongly recommended:

1. modification of the class docstring mentioning this hack, perhaps in the `Notes` section. See the discussion with @fkiraly [above](https://github.com/sktime/sktime/pull/6441/files#r1605996897).
2. addition of a test case, where you can reuse the bug report provided in the linked issue

**Comment by fkiraly** [comment]
Merging for release. Since not blocking, the updates to docstring can be made in separate PR.

**Pull request merged**